android/src/main/java/com/reactnativecommunity/netinfo/NetInfoModuleImpl.java

```diff
@@ -56,11 +56,15 @@ public void onAmazonFireDeviceConnectivityChanged(boolean isConnected) {
 
 
     public void addListener(String eventName) {
-        // iOS only
+        numberOfListeners++;
+        mConnectivityReceiver.hasListener = true;
     }
 
 
     public void removeListeners(double count) {
-        // iOS only
+        numberOfListeners -= count;
+        if (numberOfListeners == 0) {
```

Comment:
unrelated to this PR - `removeListeners` doesn't get called as we're initialising a [single listener](https://github.com/react-native-netinfo/react-native-netinfo/blob/master/src/internal/state.ts#L30) and managing emitters on JS. I think it'll be only called on [tearDown](https://github.com/react-native-netinfo/react-native-netinfo/blob/master/src/internal/state.ts#L131). I don't think it's a big issue, just wanted to point it out.